### PR TITLE
Update hardware & book related item groups

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Balance "Rearranges and rebalances item groups governing hardware spawning."```

#### Purpose of change
Item groups are currently a bit of a mess, scattered amongst many map files, and lacking comments about intended usage. This leads to things getting scattered into weird item groups, and significant item group duplication across multiple files, which makes it in turn very difficult to add new items or adjust item spawning reasonably.

#### Describe the solution
Fully fixing this problem is a huge undertaking, but I'm looking at a rational way to do it using hardware as an example.
1. Item groups that contain a wide variety of items should be broken into separate groups, which are then called by the master item group. See what I've done with the hardware and mischw item groups.
2. Item groups should be moved out of the item_groups file and into separate files. This will be tricky as currently we don't have a good way to sort those separate files... but by doing step 1, it gets easier.
3. Item groups in mapgen files should be moved to the item_groups file folders. It's too difficult to keep isolated map gen files updated.
4. Item groups meant to spawn bulk items should use the "count" flag more often.
5. Items that should spawn together should have subset "distribution" and/or "collection" flags. See what I have done with oxyacetylene torches in tools_mechanic to make them spawn alongside extra tanks, for example.

#### Additional context
- [x] Rework basic hardware, mischw, plumbing, and lumberyard groups and move them to appropriate files
- [x] Explore mapgen files for other hardware file entries and redesign them
- [x] Move books as well since I wound up dealing with a lot of them.
- [x] (ongoing) adjust some item spawns to make sure that things are spawning where they should - no anvils in home depot, but add toolboxes to jobsites, eg.
- [x] Delete instances of plumbing_gear and elec_parts (deprecated groups)

### CODE REVIEW

File: data/json/itemgroups/supplies.json [hunk 1]
```diff
@@ -22,6 +22,86 @@
...
+      { "item": "rebar", "prob": 20, "count": [ 5, 10 ] },
+      { "item": "brick", "prob": 20, "count": [ 5, 10 ] },
+      [ "superglue", 30 ],
+      [ "chain", 20 ],
+      [ "rope_6", 4 ],
+      [ "rope_30", 35 ],
+      [ "circsaw_blade", 8 ],
+      [ "glass_sheet", 5 ],
+      { "item": "pipe", "prob": 20, "count": [ 5, 10 ] },
+      { "item": "nail", "prob": 60, "count": [ 200, 400 ] },
```
Comment (Reclusive-reptile) [inline data/json/itemgroups/supplies.json]: A single "count" of nails is 100 nails. A count of 200 here will spawn 20,000 (!) nails.

File: data/json/itemgroups/supplies.json [hunk 2]
```diff
@@ -22,6 +22,86 @@
...
+    "id": "lumberyard",
+    "subtype": "collection",
+    "items": [
+      {
+        "distribution": [
+          { "item": "2x4", "prob": 500, "count": [ 20, 50 ] },
+          { "item": "log", "prob": 200, "count": [ 5, 10 ] },
+          { "item": "stick_long", "prob": 100, "count": [ 10, 20 ] },
+          { "item": "stick", "prob": 50, "count": [ 5, 10 ] },
+          { "item": "nail", "prob": 100, "count": [ 500, 1000 ] }
```
Comment (Reclusive-reptile) [inline data/json/itemgroups/supplies.json]: This will spawn a minimum of 50k nails.

File: data/json/itemgroups/tools.json [hunk 3]
```diff
@@ -253,5 +323,32 @@
...
+    "type": "item_group",
+    "items": [
+      [ "chainsaw_off", 10 ],
+      [ "elec_chainsaw_off", 10 ],
+      [ "trimmer_off", 40 ],
+      [ "jackhammer", 50 ],
+      [ "elec_jackhammer", 50 ],
+      [ "circsaw_off", 10 ]
+    ]
+  },
```
Comment (ifreund) [inline data/json/itemgroups/tools.json]:
```suggestion
  }
```

**Comment by I-am-Erk** [comment]
I am still tweaking this and need to test it in game, but I don't forsee any more major changes. Taking it off 'draft' status.

File: data/json/itemgroups/collections_trades.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  {
+    "type": "item_group",
+    "id": "hardware_plumbing",
+    "items": [ { "group": "plumbing_clothing", "prob": 100 }, { "group": "tools_plumbing", "prob": 100 }, { "group": "supplies_plumbing", "prob": 50 }, [ "survnote", 1 ] ]
+  },
+  {
+    "type": "item_group",
+    "id": "plumber_gear",
+    "//": "deprecated, use 'hardware_plubming' instead. Find and remove all examples of this group and then kill this one.",
```
Comment (Night-Pryanik) [inline data/json/itemgroups/collections_trades.json]:
```suggestion
    "//": "deprecated, use 'hardware_plumbing' instead. Find and remove all examples of this group and then kill this one.",
```

**Comment by zombiethoughts** [comment]
Just a quick mention. I was looting a public works and found 80300 nails in one spot. Yep, 80300. That's a lot of nails.

**Comment by I-am-Erk** [comment]
Do you mean you were playtesting these settings, or you found that there in the current game?

Repeat settings were fixed lately and something might have gone haywire in the public works. I can check

**Comment by Night-Pryanik** [comment]
> Just a quick mention. I was looting a public works and found 80300 nails in one spot. Yep, 80300. That's a lot of nails.

Can confirm.
изображение

I guess the culprit is here: https://github.com/CleverRaven/Cataclysm-DDA/blob/ea190d32b1042d4715bee1046278fb2b03adede7/data/json/mapgen/public_works.json#L143-L144

The `amount` field here is not total amount of nails, but instead amount of 200-charges sets of nails.

**Comment by zombiethoughts** [comment]
I'm on 8734. Had just uncovered this area of the map. On a current game.

**Comment by I-am-Erk** [comment]
Public works nail spawns aren't related to this PR, so I fixed them here: https://github.com/CleverRaven/Cataclysm-DDA/pull/29467

**Comment by kevingranade** [comment]
Looks like data/json/mapgen/house/house_vacant.json and data/json/mapgen/abandoned_barn.json still reference raw_materials.

**Comment by I-am-Erk** [comment]
thanks for the heads up

File: data/json/mapgen/homeimprovement_superstore_new.json [hunk 5]
```diff
@@ -1,120 +1,4 @@
...
-      [ "b_paint", 45 ],
-      [ "w_paint", 45 ],
-      [ "g_paint", 45 ],
-      [ "p_paint", 45 ],
-      [ "y_paint", 45 ],
-      [ "paint_brush", 55 ]
-    ]
-  },
-  {
-    "id": "carpet",
```
Comment (kevingranade) [inline data/json/mapgen/homeimprovement_superstore_new.json]: looks like you nuked the carpet item group in favor of the rug item group, but didn't switch the references to the carpet group over.

Comment (I-am-Erk) [reply data/json/mapgen/homeimprovement_superstore_new.json]: So I did. I actually didn't notice they were different.

**Comment by I-am-Erk** [comment]
OK. Runs in playtime and hardware stores seem to be balancing out okay. I think it's good to go

**Comment by ZhilkinSerg** [comment]
Obsolete `plumber_gear` item group is still used in a `More_Locations` and `Urban_Development` mods:

```
Search "plumber_gear" (29 hits in 29 files)
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_4blc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_4blc_02.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_4brc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_4tlc_00.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_4tlc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_4trc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_ublc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_ubrc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_utlc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\bandit_tower\bandit_tower_utrc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_5blc_01.json (1 hit)
	Line 56:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_5brc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_5tlc_00.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_5tlc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_5trc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_ublc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_ubrc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_utlc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\refugee_tower\refugee_tower_utrc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_4blc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_4brc_01.json (1 hit)
	Line 56:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_4tlc_00.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_4tlc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_4trc_01.json (1 hit)
	Line 55:                 "L": [ { "item": "mechanics", "chance": 70 }, { "item": "plumber_gear", "chance": 30 } ]
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_ublc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_ubrc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_utlc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\More_Locations\tri_tower\tri_tower_utrc_01.json (1 hit)
	Line 52:                 "x": [ { "item": "plumber_gear", "chance": 20 }, { "item": "hardware_bulk", "chance": 20 } ],
  C:\Projects\Cataclysm-DDA\data\mods\Urban_Development\building_jsons\urban_39_market_subway_newspaper.json (1 hit)
	Line 79:         "L": { "item": "plumber_gear", "chance": 50 },
```

**Comment by I-am-Erk** [comment]
Ah, mods. Of course.

**Pull request merged**